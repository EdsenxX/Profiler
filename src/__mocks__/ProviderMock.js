import React from 'react';
import { createStore } from 'redux';
import { Router} from 'react-router-dom';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import reducers from '../reducers';

const store = createStore(reducers, {});
const history = createBrowserHistory();

const ProviderMock = props => (
  <Provider store={store}>
    <Router location={history.location} navigator={history}>
      {props.children}
    </Router>
  </Provider>
)

export default ProviderMock;