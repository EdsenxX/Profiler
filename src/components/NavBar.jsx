// Dependencies
import { Link } from "react-router-dom";

const NavBar = () => {
  return (
    <nav className="text-white flex p-5 bg-primary-100 items-center shadow-md shadow-secondary-100/30 justify-between">
      <h1 className="font-title text-4xl">Profiler</h1>
      <div className="flex gap-4 text-lg">
        <Link to="/">Home</Link>
        <Link to="/about">About</Link>
      </div>
    </nav>
  );
};

export default NavBar;
