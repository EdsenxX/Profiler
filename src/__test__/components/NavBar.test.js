import React from "react";
import { shallow, mount } from "enzyme";
import NavBar from "../../components/NavBar";
import ProviderMock from "../../__mocks__/ProviderMock";
import {create} from "react-test-renderer";

describe("NavBar", () => {
  test("Render del componente NavBar", () => {
    const navbar = shallow(
      <ProviderMock>
        <NavBar />
      </ProviderMock>
    );
    expect(navbar.length).toEqual(1);
  });

  test("Render del titulo", () => {
    const navbar = mount(
      <ProviderMock>
        <NavBar />
      </ProviderMock>
    );
    expect(navbar.find("#Navbar-title").text()).toEqual("Profiler");
  });

});

describe ("NavBar snapshot", () => {
  test("Comprobar la UI del componente NavBar", () => {
    const navbar = create(
      <ProviderMock>
        <NavBar />
      </ProviderMock>
    );
    expect(navbar.toJSON()).toMatchSnapshot();
  });
});
