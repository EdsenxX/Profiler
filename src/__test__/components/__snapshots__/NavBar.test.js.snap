// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar snapshot Comprobar la UI del componente NavBar 1`] = `
<nav
  className="text-white flex p-5 bg-primary-100 items-center shadow-md shadow-secondary-100/30 justify-between"
>
  <h1
    className="font-title text-4xl"
    id="Navbar-title"
  >
    Profiler
  </h1>
  <div
    className="flex gap-4 text-lg"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </div>
</nav>
`;
